### PR TITLE
added summary of the MVP architecture discussions

### PR DESCRIPTION
**Pull request by erwin1**
## Description

This is a summary of the meetings and discussions we had in the last weeks about the MVP milestone.

Preview: https://github.com/erwin1/pyrsia/blob/MVP-arch-docs/docs/developers/pyrsia-high-level-overview.md
But note that the architecture diagram image is broken until this gets merged. (the link has to be absolute for it to work on the website)

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [x] I've requested a review  from `pyrsia/collaborators`.

### CODE REVIEW

File: docs/developers/pyrsia-high-level-overview.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+deactivate User
+```
+
+
+## Component architecture
+
+> green boxes: already available in the Demo milestone<br>
+> orange boxes: new in the MVP milestone
+
+![Pyrsia component diagram](pyrsia-node-high-level-components.png)
```
Comment (prince-chrismc) [inline docs/developers/pyrsia-high-level-overview.md:74]: Hmmm this is not support form the website side of things 🤔

File: docs/developers/pyrsia-high-level-overview.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+Docker ->> User: docker pull image:1.0<br>done
+
+deactivate User
+```
+
+
+## Component architecture
+
+> green boxes: already available in the Demo milestone<br>
+> orange boxes: new in the MVP milestone
```
Comment (prince-chrismc) [inline docs/developers/pyrsia-high-level-overview.md:72]: Please avoid raw HTML, if you'd like the line break we can do

```suggestion
> green boxes: already available in the Demo milestone
>
> orange boxes: new in the MVP milestone
```

File: docs/developers/pyrsia-high-level-overview.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+Pyrsia aims to support these use cases in the MVP milestone:
+
+1. An authoritative node publishes all official Docker images to a p2p network, in such a way that other nodes can retrieve and verify the authenticity of those images.
+2. A Docker client can use a locally installed Pyrsia node to pull Docker images from the Pyrsia p2p network and verify the authenticity of those images.
+3. A user can query the transparency log for any downloaded Docker image.
+
+## Sequence diagrams
+
+## Docker image publication
```
Comment (prince-chrismc) [inline docs/developers/pyrsia-high-level-overview.md:13]: For the website

```suggestion
### Docker image publication
```

File: docs/developers/pyrsia-high-level-overview.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+Admin ->> Node: Trigger publication of each official image
+
+Node ->> DockerHub: retrieve official image
+Node -->> Node: Calculate hash and<br>create transaction payload
+Node ->> PNW: Add transaction to the blockchain
+
+Node ->> PNW: Publish artifact
+```
+
+## Docker image retrieval
```
Comment (prince-chrismc) [inline docs/developers/pyrsia-high-level-overview.md:32]: For the website

```suggestion
### Docker image retrieval
```

**Review by prince-chrismc — COMMENTED** [review]
Couple suggestions so it looks better on the website... hopefully #692 will help point those problems out 🤞

File: docs/developers/pyrsia-high-level-overview.md (hunk 1 above)
Comment (prince-chrismc) [reply docs/developers/pyrsia-high-level-overview.md:74]: A workaround for the short term could be 

```suggestion
![Pyrsia component diagram](https://github.com/pyrsia/pyrsia/raw/main/docs/developers/pyrsia-node-high-level-components.png)
```

**Comment by erwin1** [comment]
Thanks for the quick feedback @prince-chrismc 
I've run `mdl docs` from #692 and fixed all warnings.

**Review by prince-chrismc — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
> I've run `mdl docs` from #692 and fixed all warnings.

If there are rules you thought were too strict don't hesitate to share, some of them are pretty aggressive and we can tweak them as needed

**Comment by erwin1** [comment]
> > I've run `mdl docs` from #692 and fixed all warnings.
> 
> If there are rules you thought were too strict don't hesitate to share, some of them are pretty aggressive and we can tweak them as needed

nothing out of the ordinary. 
the only annoying thing is was `MD029 - Ordered list item prefix` which default to `one` while my editor is unraveling numbered lists automatically to 1. 2. 3. 4... even if I type 1. 1. 1. 1...

File: docs/developers/pyrsia-high-level-overview.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+deactivate User
+```
+
+## Component architecture
+
+> green boxes: already available in the Demo milestone \
+> orange boxes: new in the MVP milestone
+
+![Pyrsia component diagram](https://github.com/pyrsia/pyrsia/raw/main/docs/developers/pyrsia-node-high-level-components.png)
```
Comment (fishseabowl) [inline docs/developers/pyrsia-high-level-overview.md:76]: https://github.com/pyrsia/pyrsia/raw/main/docs/developers/pyrsia-node-high-level-components.png
Error 404 not found.

Comment (prince-chrismc) [reply docs/developers/pyrsia-high-level-overview.md:76]: It wont work until after it's merged, thats my workaround for the website

File: docs/developers/pyrsia-high-level-overview.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+1. A user can query the transparency log for any downloaded Docker image.
+
+## Sequence diagrams
+
+### Docker image publication
+
+```mermaid
+sequenceDiagram
+participant Admin as Admin User
+participant Node as Pyrsia Central Node
```
Comment (fishseabowl) [inline docs/developers/pyrsia-high-level-overview.md:21]:
```suggestion
participant Node as Pyrsia Authoritative Node
```

File: docs/developers/pyrsia-high-level-overview.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+1. Using the Pyrsia CLI a user can query status information and transparency
+   log information
+1. Artifact service uses local storage as a first step to find requested images
+1. Artifact service uses the Pyrsia p2p network if the image is not found locally
+1. a. Artifact service uses the transparency log service to add an artifact
+   b. Artifact service uses the transparency log service to verify an artifact
+1. Pyrsia CLI uses Transparency log to find requested information / get logs
+   for an artifact
+1. Transparency log uses Blockchain to (a) add blocks and (b) listen for (new)
+   blocks
```
[the remark this answers is not in the document]
Comment (erwin1) [reply docs/developers/pyrsia-high-level-overview.md:90]: changed your suggestion for line length

**Review by prince-chrismc — APPROVED** [review]

**Review by fishseabowl — APPROVED** [review]
LGTM 🚀

**Pull request merged**